lib/galaxy_test/api/test_jobs.py

```diff
@@ -937,9 +937,97 @@ def test_job_build_for_rerun_list_list(self, history_id):
+    #     job_id = search_response.json()[0]["id"]
+    #     # delete the job without message
+    #     delete_job_response = self._delete(f"jobs/{job_id}")
+    #     self._assert_status_code_is(delete_job_response, 200)
+    #     # TODO the job is finished as such it is not stopped Think of another way to test it
+    #     assert delete_job_response.json() == True
+    #     # now that we deleted the job we should not find it anymore
+    #     search_payload = self._search_payload(history_id=history_id, tool_id="cat1", inputs=inputs)
+    #     empty_search_response = self._post("jobs/search", data=search_payload, json=True)
+    #     self._assert_status_code_is(empty_search_response, 200)
+    #     assert len(empty_search_response.json()) == 0
+
+    @pytest.mark.require_new_history
+    def test_destination_params(self, history_id):
+        dataset_id = self.__history_with_ok_dataset(history_id)
+        inputs = json.dumps({"input1": {"src": "hda", "id": dataset_id}})
+        search_response = self._create_and_search_job(history_id, inputs, tool_id="cat1")
+        job_id = search_response.json()[0]["id"]
+        destination_params_response = self._get(f"/api/jobs/{job_id}/destination_params", admin=True)
+        self._assert_status_code_is(destination_params_response, 200)
+        # TODO maybe extend test?
+
+    @pytest.mark.require_new_history
+    def test_job_metrics(self, history_id):
+        dataset_id = self.__history_with_ok_dataset(history_id)
+        inputs = json.dumps({"input1": {"src": "hda", "id": dataset_id}})
+        search_response = self._create_and_search_job(history_id, inputs, tool_id="cat1")
+        job_id = search_response.json()[0]["id"]
+        metrics_by_job_response = self._get(f"/api/jobs/{job_id}/metrics", data={"hda_ldda": "hda"})
+        self._assert_status_code_is(metrics_by_job_response, 200)
+        metrics_by_dataset_response = self._get(f"/api/datasets/{dataset_id}/metrics", data={"hda_ldda": "hda"})
+        self._assert_status_code_is(metrics_by_dataset_response, 200)
+        # TODO maybe extend test?
+
+    @pytest.mark.require_new_history
+    def test_parameters_display(self, history_id):
+        dataset_id = self.__history_with_ok_dataset(history_id)
+        inputs = json.dumps({"input1": {"src": "hda", "id": dataset_id}})
+        search_response = self._create_and_search_job(history_id, inputs, tool_id="cat1")
+        job_id = search_response.json()[0]["id"]
+        display_parameters_by_job_response = self._get(
+            f"/api/jobs/{job_id}/parameters_display", data={"hda_ldda": "hda"}
+        )
+        self._assert_status_code_is(display_parameters_by_job_response, 200)
+        display_parameters_by_dataset_response = self._get(
+            f"/api/datasets/{dataset_id}/parameters_display", data={"hda_ldda": "hda"}
+        )
+        self._assert_status_code_is(display_parameters_by_dataset_response, 200)
+        # TODO maybe extend test
```

Comment:
So the endpoints "destination_params", "metrics" and "parameters_display" had no tests yet. I wrote tests for all of them, but they are really simple. Basically the only check that the endpoint works. It does not check the return data, as I do not really know anything about that.
Maybe they should be extended, any suggestions on if and how?

Reply:
These are fairly internal, I think this is good enough for now!